help/Readme.md

```diff
@@ -0,0 +1,116 @@
+### General
+Virtual-environment project uses packer tool and contains a number of templates for several OS types: Windows 2016/2019, Ubuntu 16.04/18.04/20.04.
+Each image template contains set of scripts that will be executed on virtual machine.
+Packer process will initialize connection to Azure subscription via Azure CLI, and deploy required resources, such as temporary resource group, network interfaces and VM from the "clean" image that specified in template.
+If deployment succeeded, build agent will connect to the deployed VM and begin to execute installation steps from selected the template.
+Connection can be established via SSH or WinRM, depending on image type, so the ports between build agent and temporary VM have to be opened for connection.
+If any template step fails, image generation will be aborted and temporary VM will be terminated.
+If all steps are succeeded, a temporary VM snapshot will be converted to VHD and then uploaded to the specified Azure Storage Account. After that image can be deployed further from this account.
+
+### Build Agent requirements
+- `OS` - Windows/Linux
+- `packer` - Can be downloaded from https://www.packer.io/downloads
+- `PowerShell 5.0 or higher` or `PSCore` for linux distributes.
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to temporary VM packer will use WinRM, or SSH connections on public IP interfaces.
+If you use build agent located in some Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case if firewall restrictions forbid connections from public addresses, you may deploy private virtual network resources, and pass them as arguments to packer, so temporary virtual machines can use private connections inside Vlan. 
```

Comment:
```
#### Azure DevOps self-hosted pool requirements
To connect to a temporary VM packer use WinRM or SSH connections on public IP interfaces.
If you use a build agent located in an Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
In case of firewall restrictions, prohibiting connections from public addresses, private virtual network resources can be deployed and passed as arguments to the packer. This approach allows virtual machines to use private connections inside VLAN. 
```

Reply:
Thanks. Fixed.